bundles/org.eclipse.equinox.preferences/src/org/eclipse/core/internal/preferences/PreferencesService.java

```diff
@@ -149,8 +144,10 @@ public IStatus applyPreferences(IExportedPreferences preferences) throws CoreExc
 						}
 					}
 				}
+				if (!propsToRemove.isEmpty() && !(globalNode instanceof EclipsePreferences)) {
+					propsToRemove.replaceAll(String::intern);
+				}
```

Comment:
The elements in the keys array above are obtained from an `ExportedPreferences`, which is a subclass of `EclipsePreference` and therefore has its keys already interned (unless somebody extends this internl class and changes the method to store elements, but those that do that should take care of interning themself).

On the other hand `globalNode` is just a `IEclipsePreferences` node from the tree and because `Clients may implement this interface`, it can be an arbitrary implementation. Therefore interning  `propsToRemove` can only be skipped if the node is of a `EclipsePreferences` subclass.